src/v/kafka/client/client.cc

```diff
@@ -264,4 +265,18 @@ ss::future<offset_commit_response> client::consumer_offset_commit(
       });
 }
 
+ss::future<kafka::fetch_response> client::consumer_fetch(
+  const group_id& g_id,
+  const member_id& m_id,
+  std::chrono::milliseconds timeout,
+  int32_t max_bytes) {
+    auto end = model::timeout_clock::now() + timeout;
+    return gated_retry_with_mitigation([this, g_id, m_id, end, max_bytes]() {
+        return get_consumer(g_id, m_id)
+          .then([end, max_bytes](shared_consumer_t c) {
+              return c->fetch(end - model::timeout_clock::now(), max_bytes);
```

Comment:
what happens if enough time elapsed that `end - now` becomes negative, or is that not possible? it looks technically possible.